appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -67,48 +68,68 @@ public void read(String documentId, String partition, ReadableMap readOptionsMap
-                jsDocumentWrapper.putString(JSON_VALUE_KEY, documentWrapper.getJsonValue());
-
-                // Pass milliseconds back to JS object since `WritableMap` does not support `Date` as values
-                jsDocumentWrapper.putDouble(LAST_UPDATED_DATE_KEY, documentWrapper.getLastUpdatedDate().getTime());
-                jsDocumentWrapper.putBoolean(IS_FROM_DEVICE_CACHE_KEY, documentWrapper.isFromDeviceCache());
-
-                if (deserializedValue.isJsonPrimitive()) {
-                    JsonPrimitive jsonPrimitive = deserializedValue.getAsJsonPrimitive();
-                    if (jsonPrimitive.isString()) {
-                        jsDocumentWrapper.putString(DESERIALIZED_VALUE_KEY, jsonPrimitive.getAsString());
-                    } else if (jsonPrimitive.isNumber()) {
-                        jsDocumentWrapper.putDouble(DESERIALIZED_VALUE_KEY, jsonPrimitive.getAsDouble());
-                    } else if (jsonPrimitive.isBoolean()) {
-                        jsDocumentWrapper.putBoolean(DESERIALIZED_VALUE_KEY, jsonPrimitive.getAsBoolean());
-                    }
-                } else if (deserializedValue.isJsonObject()) {
-                    JsonObject jsonObject = deserializedValue.getAsJsonObject();
-                    WritableMap writableMap = AppCenterReactNativeDataUtils.convertJsonObjectToWritableMap(jsonObject);
-                    jsDocumentWrapper.putMap(DESERIALIZED_VALUE_KEY, writableMap);
-                } else if (deserializedValue.isJsonArray()) {
-                    JsonArray jsonArray = deserializedValue.getAsJsonArray();
-                    WritableArray writableArray = AppCenterReactNativeDataUtils.convertJsonArrayToWritableArray(jsonArray);
-                    jsDocumentWrapper.putArray(DESERIALIZED_VALUE_KEY, writableArray);
-                } else {
-                    jsDocumentWrapper.putNull(DESERIALIZED_VALUE_KEY);
+        Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer("Read failed", promise));
+    }
+
+    @ReactMethod
+    public void create(String documentId, ReadableMap readableMap, String partition, ReadableMap writeOptionsMap, final Promise promise) {
+        WriteOptions writeOptions;
+        if (writeOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
+            writeOptions = new WriteOptions(writeOptionsMap.getInt(TIME_TO_LIVE_KEY));
+        } else {
+            writeOptions = new WriteOptions(TimeToLive.DEFAULT);
+        }
+        JsonObject jsonObject = AppCenterReactNativeDataUtils.convertReadableMapToJsonObject(readableMap);
+        Data.create(documentId, jsonObject, JsonElement.class, partition, writeOptions).thenAccept(new Consumer("Create failed", promise));
+    }
+
+    private class Consumer implements AppCenterConsumer<DocumentWrapper<JsonElement>> {
+        private Promise promise;
+        private String errorCode;
+
+        private Consumer(String errorCode, final Promise promise) {
```

Comment:
```suggestion
        private Consumer(String errorCode, Promise promise) {
```

Reply:
We also put new lines between fields, and on Android we usually use mField convention (and thus not `this`).

Reply:
Changed